pkg/cloudprovider/registry.go

```diff
@@ -0,0 +1,35 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudprovider
+
+import (
+	v1 "k8s.io/api/core/v1"
+)
+
+type ResourceFlags byte
+
+const (
+	// ResourceFlagNone is the default settings for resources where we normally scale up to larger nodes with no
+	// restrictions.
+	ResourceFlagNone ResourceFlags = iota
+
+	// ResourceFlagMinimizeUsage indicates that instances which have this resource should not be considered for node
+	// creation unless a pod specifically requests this resource type.  This is useful for preventing non-GPU workloads
+	// from possibly scheduling to more expensive GPU instance type, or from causing a GPU instance type to scale up to
+	// the next larger type due to a non-GPU workload
+	ResourceFlagMinimizeUsage
```

Comment:
I keep thinking about this complexity. I'm wondering if it's enough to purely delineate on WellKnown vs NotWellKnown resource types? If we know about it (e.g., CPU, Memory, EphemeralStorage) then we can use the `ResourceFlagNone` behavior. If we don't know about it, we can use `ResourceFlagMinimizeUsage` behavior. This will enable us to simply not configure any of this (for now).

My guess is that we will need to be more sophisticated with binpacking in the future, but I don't think we have enough use cases to have clarity on what the parameters are. Given that with currently known resource types, the above suggestion is equivalent in implementation.

Happy to move forward in either direction.

Reply:
I left a longer comment below, but I'm starting to think this complexity isn't worth it since kube-scheduler won't follow this logic and we're left with using taints/tolerations again which are already built-in.